xla/pjrt/gpu/se_gpu_pjrt_client.cc

```diff
@@ -1006,6 +1007,20 @@ absl::Status BuildDistributedDevices(
         TF_RET_CHECK(it->second != nullptr);
         local_device = std::move(it->second);
         gpu_device_ids[device_proto.local_device_ordinal()] = global_device_id;
+        // Assign some descriptive names for profiling tools.
+        auto suffix =
+            absl::StrFormat(":#global=%d,local=%d,process=%d,slice=%d#",
+                            device_proto.global_device_id(),
+                            device_proto.local_device_ordinal(), node.node_id(),
+                            device_proto.slice_index());
+        // Name the device.
+        tsl::profiler::NameDevice(device_proto.local_device_ordinal(),
+                                  ("Xla" + suffix).c_str());
+        // Name the thread that launches work on this device.
+        local_device->execute_thread()->Schedule(
```

Comment:
I believe it would be possible to tell the profiler/NVTX implementation about the name used in the Env interface.

As it stands, it would require a more significant reorganisation of the XLA code to ensure that an informative name is available at thread creation time. The existing names are not very detailed or informative. Alternatively, it would require a new API to set the name later than thread creation time.

Reply:
Please expand the comment and document why the name cannot be set at creation time and needs to be overridden here.

Reply:
Done